collect.py

```diff
@@ -71,43 +76,77 @@ def store_results(temppath):
     print('Results stored in %s.' % fname)
 
 
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '-m', '--model',
+        help='Model to operate in. Accepts <controller name>:<model name>')
+    return parser.parse_args()
+
+
 def main():
+    options = parse_args()
+    if not options.model:
+        model = check_output(['juju', 'switch'])
+        model = model.decode()
+        if len(model.split(':')) != 2:
+            print("juju controller:model unknown")
+            sys.exit(1)
+    else:
+        model = options.model
+        if len(model.split(':')) != 2:
+            print("juju controller:model unknown")
+            sys.exit(1)
+
+    model = "-m {}".format(model)
+
     tempdir = tempfile.TemporaryDirectory()
     temppath = os.path.join(tempdir.name, 'results')
     os.makedirs(temppath)
 
     print('Getting juju status...')
     try:
-        raw_status = check_output('juju status --format json'.split())
+        raw_status = check_output(
+            'juju status {} --format json'.format(model).split())
         status = json.loads(raw_status.decode())
     except:
         print('Error getting juju status. Aborting.')
         return
 
-    debug_action(temppath, status, 'kubernetes-master')
-    debug_action(temppath, status, 'kubernetes-worker')
-    debug_action(temppath, status, 'etcd')
+    debug_action(temppath, status, model, 'kubernetes-master')
+    debug_action(temppath, status, model, 'kubernetes-worker')
+    debug_action(temppath, status, model, 'etcd')
     # FIXME: no debug action on kubeapi-load-balancer, easyrsa, flannel
 
-    command(temppath, 'status', 'juju status --format yaml')
-    command(temppath, 'debug-log', 'juju debug-log --replay')
-    command(temppath, 'model-config', 'juju model-config')
-    command(temppath, 'controller-debug-log', 'juju debug-log -m controller --replay')
-    command(temppath, 'storage', 'juju storage --format yaml')
-    command(temppath, 'storage-pools', 'juju storage-pools --format yaml')
-    command(temppath, 'kubernetes-master-config', 'juju config kubernetes-master --format yaml')
-    command(temppath, 'kubernetes-worker-config', 'juju config kubernetes-worker --format yaml')
-    command(temppath, 'kubeapi-load-balancer-config', 'juju config kubeapi-load-balancer --format yaml')
-    command(temppath, 'etcd-config', 'juju config etcd --format yaml')
-    command(temppath, 'easyrsa-config', 'juju config easyrsa --format yaml')
-    command(temppath, 'flannel-config', 'juju config flannel --format yaml')
+    command(temppath, 'status', 'juju status {} --format yaml'.format(model))
+    command(temppath, 'debug-log', 'juju debug-log {} --replay'.format(model))
+    command(temppath, 'model-config', 'juju model-config {}'.format(model))
+    command(temppath, 'controller-debug-log',
+            'juju debug-log -m {}:controller --replay'.format(
+                model.split(':')[0]))
```

Comment:
Oops, this failed in my test run:

```
$ cat controller-debug-log.err
ERROR unrecognized args: ["aws-dev:controller"]
```

I guess because the command was something like `juju debug-log -m -m aws-dev:controller`? I think if you just remove the `-m` from this line then we should be good to go.